ee/WFCORE-4135_Support_for_ read_only_server_config_dir.adoc

```diff
@@ -0,0 +1,94 @@
+= Support for read only server configuration directory
+:author:            Yeray Borges
+:email:             yborgess@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+The following settings allow the server to use a read only configuration file:
+
+* Standalone mode:
+** `--read-only-server-config=<server config file name>`
+* Domain mode:
+** `--read-only-server-config=<domain config file name>`
+** `--read-only-host-config=<host config file name>`
+
+The point of these settings is the configuration you want is locked down. Any change will not be saved upon a server restart, however it is possible to apply runtime changes that require a server reload. These settings are useful if the user has an approved configuration that must not be changed but he wants to test a different configuration without worring about the possibility to make the changes permanent.
+
+However, neither of these configurations allow the server to start when the whole configuration directory is read only. During the booting phase the server needs write access to the configuration directory, for example, to store the history backups or temporal files.
+
+History backups are senseless running on a container, and certain cloud use cases could require a non-writable storage mounted as a configuration directory, for example, by default config maps are mounted as https://github.com/kubernetes/kubernetes/pull/58720[read only] volumes and a common pattern is to supply the server the configuration directory by using a ConfigMap. When we are running on a non-containerized environment, we could have very strict security rules that could force us the use of a read only configuration directory.
+
+The goal of this feature is to make `--read-only-server-config=<server config file name>` setting work in standalone mode even if the whole configuration directory pointed by `jboss.server.config` is a read only directory.
+
+Domain mode changes are not initially included on this feature.
+
+== Required changes for cloud images
+
+The cloud images should be smart enough to detect if the server configuration directory is a read only directory, and in this case, the launch command for the underlying server should incorporate the `--read-only-server-config=<server config file name>` argument to make it possible to start the server under this circumstance.
```

Comment:
This should be captured in _Hard Requirements_

Reply:
Thanks for reviewing @OndrejKotek, I've just added them as hard requirements.